packages/utils-dom/src/node.ts

```diff
@@ -0,0 +1,121 @@
+import { HTMLDocument } from './htmldocument';
+import { NodeData } from './types';
+
+export class Node {
+    private _node: NodeData;
+
+    public ownerDocument: HTMLDocument;
+
+    public constructor(data: NodeData, ownerDocument: HTMLDocument) {
+        this._node = data;
+        this.ownerDocument = ownerDocument;
+    }
+
+    private get _owner() {
+        return this.ownerDocument || this;
+    }
+
+    public get childNodes(): Node[] {
+
+        const result: Node[] = [];
+
+        if ('children' in this._node) {
+            for (const child of this._node.children) {
+                result.push(this._owner.getNodeFromData(child));
+            }
+        }
+
+        return result;
+    }
+
+    public get nodeName(): string {
+        switch (this._node.type) {
+            case 'comment':
+                return '#comment';
+            case 'directive':
+                return this._node.nodeName || this._node['x-name'] || '';
+            case 'root':
+                return '#document';
+            case 'script':
+            case 'style':
+            case 'tag':
+                return this._node.name.toUpperCase();
+            case 'text':
+                return '#text';
+            /* istanbul ignore next */
+            default:
+                throw new Error(`Unrecognized node type ${(this._node as any).type}`);
+        }
+    }
+
+    public get nodeType(): number {
```

Comment:
Will be good to point to the spec just for reference

Reply:
I'd prefer to reference MDN (which itself links to the related spec or specs). I find it to be a better place to start to understand the basics of what an API does. What do you think?

E.g. for this case the link would be
https://developer.mozilla.org/docs/Web/API/Node/nodeType

Reply:
SGTM!